internal/datasource/host/process.go

```diff
@@ -8,12 +8,13 @@ package host
 import (
 	"context"
 	"log/slog"
+	"strings"
 
 	"github.com/nginx/agent/v3/internal/model"
 	"github.com/shirou/gopsutil/v3/process"
 )
 
-func GetProcesses(ctx context.Context) ([]*model.Process, error) {
+func GetProcesses(ctx context.Context) (map[int32]*model.Process, error) {
```

Comment:
I would propose we either add a comment here explaining what the key of our map is (i.e. the PID) or then we add a clarifying type alias for the maps key:
```go
type PID int32 // alternatively could call it 'ProcessID'

func GetProcesses(ctx context.Context) (map[PID]*model.Process, error) 
```

Reply:
Also, do we want to rename this to `GetNginxProcesses` now?

Reply:
To the first comment, yes I'll add that I meant to and it slipped my mind.
Second comment, I can ya I left it as is so we could in future add checks for other processes but can always be changed again later when we add those in.

Reply:
TBH I think we should use the type alias, as we are passing this map thoughout our code base and I think it significantly improves readability.

Reply:
Agreed, I'll add an alias

Reply:
> Second comment, I can ya I left it as is so we could in future add checks for other processes but can always be changed again later when we add those in.

Makes sense 👍